assets/src/setup/components/reader-themes-context-provider.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * WordPress dependencies
+ */
+import { createContext, useEffect, useState, useRef, useContext, useMemo } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+/**
+ * Internal dependencies
+ */
+import { Options } from './options-context-provider';
+
+export const ReaderThemes = createContext();
+
+/**
+ * Context provider for options retrieval and updating.
+ *
+ * @param {Object} props Component props.
+ * @param {string} props.ajaxurl WP AJAX URL.
+ * @param {?any} props.children Component children.
+ * @param {string} props.readerThemesEndpoint REST endpoint to fetch reader themes.
+ * @param {string} props.updatesNonce Nonce for the AJAX request to install a theme.
+ */
+export function ReaderThemesContextProvider( { ajaxurl, children, readerThemesEndpoint, updatesNonce } ) {
```

Comment:
Even though the global variable is called `ajaxurl` we could apply our naming convention here. Also maybe prepend `wp` to indicate it's WordPress specific.

```suggestion
export function ReaderThemesContextProvider( { wpAjaxUrl, children, readerThemesEndpoint, updatesNonce } ) {
```